File: Ghpr.CouchDb/Mappers/SimpleItemInfoDtoMapper.cs
using Ghpr.Core.Common;
using Ghpr.CouchDb.Entities;

namespace Ghpr.CouchDb.Mappers
{
    public static class SimpleItemInfoDtoMapper
    {
        public static SimpleItemInfo MapSimpleItemInfo(this SimpleItemInfoDto simpleItemInfoDto, string itemName = "")
        {
            var simpleItemInfo = new SimpleItemInfo
            {
                ItemName = itemName.Equals("") ? simpleItemInfoDto.ItemName : itemName,
                Date = simpleItemInfoDto.Date
            };
            return simpleItemInfo;
        }

        public static SimpleItemInfoDto ToDto(this SimpleItemInfo entity)
        {
            var dto = new SimpleItemInfoDto
            {
                ItemName = entity.ItemName,
                Date = entity.Date
            };
            return dto;
        }
    }
}

File: Ghpr.Controller/controller/common/localFileSystem/mappers/SimpleItemInfoDtoMapper.ts
///<reference path="./../entities/SimpleItemInfo.ts"/>
///<reference path="./../../../dto/SimpleItemInfoDto.ts"/>

class SimpleItemInfoDtoMapper {
    static map(simpleItemInfo: SimpleItemInfo): SimpleItemInfoDto {
        let simpleItemIntoDto = new SimpleItemInfoDto();
        simpleItemIntoDto.date = simpleItemInfo.date;
        simpleItemIntoDto.itemName = simpleItemInfo.itemName;
        return simpleItemIntoDto;
    }
}